proposals/0000-ios-scenedelegate-and-resizing.md

```diff
@@ -0,0 +1,173 @@
+  * invoke RN `RCTLinkingManager` methods from `SceneDelegate`:  
+    * `scene:continueUserActivity:`
+    * `scene:openURLContexts:`
+  * update of the app's Info.plist to include a UIApplicationSceneManifest specifying the support and disabling multiple scenes capability  
+* In React Native code:  
+  * migration of code that relies on `launchOptions` and deprecated `UIApplicationLaunchOptions`\* keys to UIScene lifecycle and UIScene.ConnectionOptions.userActivities  
+* In React Native code, RN native libraries’ code:  
+  * migration of app lifecycle methods from application\* to scene\* as per [https://developer.apple.com/documentation/technotes/tn3187-migrating-to-the-uikit-scene-based-life-cycle](https://developer.apple.com/documentation/technotes/tn3187-migrating-to-the-uikit-scene-based-life-cycle)  
+  * migration of code that relies on *any* *deprecated* `AppDelegate`-related APIs, which detailed description of is presented below
+
+To enforce the user not to enable the `UIApplicationSceneManifest`.`UIApplicationSupportsMultipleScenes` capability in `Info.plist`, from now on, the following warning would be printed from `RCTReactNativeFactory` provided the user had enabled the multi-window capability:
+
+![](assets/0000-console-warning-unsupported-configuration.png)
+
+#### Migration: RN app template, RNTester
+
+In case of the RN app template, the iOS boilerplate code is limited only to basic bootstrapping of the application. This implies adjustments to:
+
+* `Info.plist` \- add support for `SceneDelegate`  
+* ``SceneDelegate`.mm` \- implement the `SceneDelegate`  
+* ``AppDelegate`.mm` \- move app bootstrap code from here to ``SceneDelegate`.mm`
+
+#### Migration: inspect usages of `RCTReactNativeFactory`
+
+Support methods for initializing React Native from ``SceneDelegate``’s lifecycle methods.
+
+#### Migration: `RCTLinkingManager`
+
+The linking manager is using ``AppDelegate`` methods for handling URLs being opened at runtime. This needs to be migrated to `SceneDelegate`. To maintain backwards compatibility, we can implement both approaches and \- to ensure that only one is invoked at a given time \- conditionally check if the app is based on ``AppDelegate`` or has scenes to ensure only one listener handles the event. The Scene lifecycle options (`NSDictionary`) are adapted to the format of ``AppDelegate`` launchOptions (`NSDictionary` as well).
+
+#### Migration: RCTDevLoadingView
+
+The does not account for updating the overlay `UIWindow` constraints so it does not get resized along with the main window. The proposed fix is to update the constraints based on KVO observation of the main window’s frame.
+
+![](assets/0000-RCTDevLoadingView-width.png)
+![](assets/0000-RCTDevLoadingView-width-clipped.png)
```

Comment:
this part might have changed a little in the meantime. Worth rechecking the status of the codebase.

Reply:
Sure, I will validate this on the newest main